src/toil/jobStores/awsJobStore.py

```diff
@@ -169,7 +173,9 @@ def delete( self, jobStoreID ):
             log.debug( "Deleting %d file(s) associated with job %s", len( items ), jobStoreID )
             for attempt in retry_sdb( ):
                 with attempt:
-                    self.versions.batch_delete_attributes( { item.name: None for item in items } )
+                    loops = int(math.ceil(float(len(items))/batch_delete_limit))
```

Comment:
Also, you shouldn't only retry the current batch. So the retry loop and the for-each batch loop need to be swapped.